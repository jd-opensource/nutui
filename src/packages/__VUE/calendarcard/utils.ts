import Utils from '@/packages/utils/date'
import { CalendarCardDay, CalendarCardMonth, CalendarCardValue } from './types'

export const convertDateToDay = (date: Date) => {
  return date
    ? {
        year: date.getFullYear(),
        month: date.getMonth() + 1,
        date: date.getDate()
      }
    : null
}

export const convertDayToDate = (day: CalendarCardDay) => {
  return day ? new Date(day.year, day.month - 1, day.date) : null
}

export const valueToRange = (val?: CalendarCardValue) => {
  if (Array.isArray(val)) {
    return val.map((date: Date) => {
      return convertDateToDay(date)
    })
  }
  return val ? [convertDateToDay(val)] : []
}

export const rangeTovalue = (range?: CalendarCardDay[]) => {
  if (Array.isArray(range)) {
    return range.map((day: CalendarCardDay) => {
      return convertDayToDate(day)
    })
  }
  return range ? [convertDayToDate(range)] : []
}

/**
 * 获取当月面板中前一个月的日期数据
 */
export const getPrevMonthDays = (year: number, month: number, firstDayOfWeek: number) => {
  let prevMonth = month - 1
  let prevYear = year
  if (prevMonth <= 0) {
    prevMonth = 12
    prevYear -= 1
  }
  let days = Utils.getMonthPreDay(year, month)
  days -= firstDayOfWeek
  if (days >= 7) {
    days -= 7
  }

  const preDates = Utils.getMonthDays(`${prevYear}`, `${prevMonth}`)
  const months = Array.from(Array(preDates), (_, k) => {
    return {
      type: 'prev',
      year: prevYear,
      month: prevMonth,
      date: k + 1
    } as CalendarCardDay
  })
  return months.slice(preDates - days)
}

/**
 * 获取当前月的日期数据
 */
export const getCurrentMonthDays = (year: number, month: number) => {
  const days = Utils.getMonthDays(`${year}`, `${month}`)
  return Array.from(Array(days), (_, k) => {
    return {
      type: 'current',
      year,
      month,
      date: k + 1
    } as CalendarCardDay
  })
}

/**
 * 根据日期获取当前周的起始日期
 */
export const getCurrentWeekDays = (day: CalendarCardDay, firstDayOfWeek: number): CalendarCardDay[] => {
  const current = new Date(day.year, day.month - 1, day.date)
  const count = (current.getDay() + 7 - firstDayOfWeek) % 7
  return [
    convertDateToDay(new Date(current.getTime() - 24 * 60 * 60 * 1000 * count)) as CalendarCardDay,
    convertDateToDay(new Date(current.getTime() + 24 * 60 * 60 * 1000 * (6 - count))) as CalendarCardDay
  ]
}

/**
 * 获取 month 月份数据
 * @param month
 * @param firstDayOfWeek
 * @returns
 */
export const getDays = (month: CalendarCardMonth, firstDayOfWeek: number) => {
  const y = month.year
  const m = month.month
  const days = [...getPrevMonthDays(y, m, firstDayOfWeek), ...getCurrentMonthDays(y, m)] as CalendarCardDay[]
  const size = days.length
  const yearOfNextMonth = month.month === 12 ? month.year + 1 : month.year
  const monthOfNextMonth = month.month === 12 ? 1 : month.month + 1
  // 补全 6 行 7 列视图
  for (let i = 1; i <= 42 - size; i++) {
    days.push({
      type: 'next',
      year: yearOfNextMonth,
      month: monthOfNextMonth,
      date: i
    })
  }
  return days
}

/**
 * 比较两个日期
 * @param day1
 * @param day2
 * @returns
 */
export const compareDay = (day1: CalendarCardDay, day2: CalendarCardDay) => {
  if (day1 && day2) {
    if (day1.year === day2.year) {
      if (day1.month === day2.month) {
        return day1.date - day2.date
      }
      return day1.month - day2.month
    }
    return day1.year - day2.year
  }
}

export const isSameDay = (day1: CalendarCardDay, day2: CalendarCardDay) => {
  return compareDay(day1, day2) === 0
}
