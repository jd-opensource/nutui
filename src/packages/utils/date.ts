const Utils = {
  /**
   * 是否为闫年
   * @return {Boolse} true|false
   */
  isLeapYear: function (y: number): boolean {
    return (y % 4 == 0 && y % 100 != 0) || y % 400 == 0;
  },

  /**
   * 返回星期数
   * @return {String}
   */
  getWhatDay: function (year: number, month: number, day: number): string {
    const date = new Date(year + '/' + month + '/' + day);
    const index = date.getDay();
    const dayNames = ['星期日', '星期一', '星期二', '星期三', '星期四', '星期五', '星期六'];
    return dayNames[index];
  },

  /**
   * 返回星期数
   * @return {Number}
   */
  getMonthPreDay: function (year: number, month: number): number {
    const date = new Date(year + '/' + month + '/01');
    let day = date.getDay();
    if (day == 0) {
      day = 7;
    }
    return day;
  },

  /**
   * 返回月份天数
   * @return {Number}
   */
  getMonthDays: function (year: string, month: string): number {
    if (/^0/.test(month)) {
      month = month.split('')[1];
    }
    return ([0, 31, this.isLeapYear(Number(year)) ? 29 : 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31] as number[])[
      month as any
    ];
  },

  /**
   * 补齐数字位数
   * @return {string}
   */
  getNumTwoBit: function (n: number): string {
    n = Number(n);
    return (n > 9 ? '' : '0') + n;
  },

  /**
   * 日期对象转成字符串
   * @return {string}
   */
  date2Str: function (date: Date, split?: string): string {
    split = split || '-';
    const y = date.getFullYear();
    const m = this.getNumTwoBit(date.getMonth() + 1);
    const d = this.getNumTwoBit(date.getDate());
    return [y, m, d].join(split);
  },

  /**
   * 返回日期格式字符串
   * @param {Number} 0返回今天的日期、1返回明天的日期，2返回后天得日期，依次类推
   * @return {string} '2014-12-31'
   */
  getDay: function (i: number): string {
    i = i || 0;
    let date = new Date();
    const diff = i * (1000 * 60 * 60 * 24);
    date = new Date(date.getTime() + diff);
    return this.date2Str(date);
  },

  /**
   * 时间比较
   * @return {Boolean}
   */
  compareDate: function (date1: string, date2: string): boolean {
    const startTime = new Date(date1.replace('-', '/').replace('-', '/'));
    const endTime = new Date(date2.replace('-', '/').replace('-', '/'));
    if (startTime >= endTime) {
      return false;
    }
    return true;
  },

  /**
   * 时间是否相等
   * @return {Boolean}
   */
  isEqual: function (date1: string, date2: string): boolean {
    const startTime = new Date(date1).getTime();
    const endTime = new Date(date2).getTime();
    if (startTime == endTime) {
      return true;
    }
    return false;
  },
  getMonthWeek: function (year: string, month: string, date: string, firstDayOfWeek: number = 0): number {
    const dateNow = new Date(Number(year), parseInt(month) - 1, Number(date));
    let w = dateNow.getDay(); //星期数
    let d = dateNow.getDate();
    let remainder = 6 - w;
    if (firstDayOfWeek !== 0) {
      w = w == 0 ? 7 : w;
      remainder = 7 - w;
    }
    return Math.ceil((d + remainder) / 7);
  },
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  getYearWeek: function (year: string, month: string, date: string, firstDayOfWeek: number = 0): number {
    const dateNow = new Date(Number(year), parseInt(month) - 1, Number(date));
    const dateFirst = new Date(Number(year), 0, 1);
    const dataNumber = Math.round((dateNow.valueOf() - dateFirst.valueOf()) / 86400000);
    return Math.ceil((dataNumber + (dateFirst.getDay() + 1 - 1)) / 7);
  },
  getWeekDate: function (year: string, month: string, date: string, firstDayOfWeek: number = 0): string[] {
    const dateNow = new Date(Number(year), parseInt(month) - 1, Number(date));
    const nowTime = dateNow.getTime();
    let day = dateNow.getDay();
    if (firstDayOfWeek === 0) {
      const oneDayTime = 24 * 60 * 60 * 1000;
      //显示周日
      const SundayTime = nowTime - day * oneDayTime; //本周的周日
      //显示周六
      const SaturdayTime = nowTime + (6 - day) * oneDayTime; //本周的周六

      const sunday = this.date2Str(new Date(SundayTime));
      const saturday = this.date2Str(new Date(SaturdayTime));
      return [sunday, saturday];
    } else {
      day = day == 0 ? 7 : day;
      const oneDayTime = 24 * 60 * 60 * 1000;
      //显示周一
      const MondayTime = nowTime - (day - 1) * oneDayTime; //本周的周一
      //显示周日
      const SundayTime = nowTime + (7 - day) * oneDayTime; //本周的周日

      const monday = this.date2Str(new Date(MondayTime));
      const sunday = this.date2Str(new Date(SundayTime));
      return [monday, sunday];
    }
  },
  formatResultDate: function (date: string) {
    let days = [...date.split('-')];
    days[2] = Utils.getNumTwoBit(Number(days[2]));
    days[3] = `${days[0]}-${days[1]}-${days[2]}`;
    days[4] = Utils.getWhatDay(+days[0], +days[1], +days[2]);
    return days;
  }
};

export default Utils;
