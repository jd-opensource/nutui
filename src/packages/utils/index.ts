export * from './useRelation';
export * from './useLocale';
