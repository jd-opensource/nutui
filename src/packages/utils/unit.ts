import { ComponentPublicInstance, nextTick } from 'vue';
import { VueWrapper, DOMWrapper } from '@vue/test-utils';

function getTouch(el: Element | Window, x: number, y: number) {
  return {
    identifier: Date.now(),
    target: el,
    pageX: x,
    pageY: y,
    clientX: x,
    clientY: y,
    radiusX: 2.5,
    radiusY: 2.5,
    rotationAngle: 10,
    force: 0.5
  };
}

export function trigger(
  wrapper: VueWrapper<ComponentPublicInstance<any, any, any>> | DOMWrapper<Element> | Element | Window,
  eventName: string,
  x = 0,
  y = 0,
  options: any = {}
) {
  const el = 'element' in wrapper ? wrapper.element : wrapper;
  const touchList = options.touchList || [getTouch(el, x, y)];

  if (options.x || options.y) {
    touchList.push(getTouch(el, options.x, options.y));
  }

  const event = document.createEvent('CustomEvent');
  event.initCustomEvent(eventName, true, true, {});

  Object.assign(event, {
    clientX: x,
    clientY: y,
    touches: touchList,
    targetTouches: touchList,
    changedTouches: touchList
  });

  el.dispatchEvent(event);

  return nextTick();
}
export async function mockScrollTop(value: number) {
  Object.defineProperty(window, 'scrollTop', { value, writable: true });
  trigger(window, 'scroll');
  return nextTick();
}

export function sleep(delay = 0): Promise<void> {
  return new Promise((resolve) => {
    setTimeout(resolve, delay);
  });
}
