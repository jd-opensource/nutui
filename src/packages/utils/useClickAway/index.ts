import { Ref, isRef, onMounted, onUnmounted } from 'vue';

type TargetType = Element | Ref<Element | undefined>;

export function useClickAway(onClickAway: () => void, target: TargetType | TargetType[], eventName = 'touchstart') {
  const handler = (event: Event) => {
    const targets = Array.isArray(target) ? target : [target];
    const inTarget = targets.some((item) => {
      const targetElement = isRef(item) ? item.value : item;
      return !targetElement || targetElement.contains(event.target as Node);
    });
    if (!inTarget) {
      onClickAway();
    }
  };

  onMounted(() => {
    window.addEventListener(eventName, handler, true);
  });

  onUnmounted(() => {
    window.removeEventListener(eventName, handler, true);
  });
}
