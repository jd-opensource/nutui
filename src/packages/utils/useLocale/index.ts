import locale from '@/packages/locale';
import { getPropByPath, isFunction } from '../util';

export const useLocale = (name = '') => {
  return (keyPath: string, ...args: unknown[]) => {
    name = name.toLocaleLowerCase();
    const languages = locale.languages();
    let text = keyPath;
    if (name && name.startsWith('nut')) {
      text = `${name.slice(3)}.${keyPath}`;
    }
    const res = getPropByPath(languages, text) || getPropByPath(languages, keyPath);
    return isFunction(res) ? res(...args) : res;
  };
};
