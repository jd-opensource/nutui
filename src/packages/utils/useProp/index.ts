import { computed, getCurrentInstance } from 'vue';
import type { ComponentPublicInstance, ComputedRef } from 'vue';

export const useProp = <T>(name: string): ComputedRef<T | undefined> => {
  const vm = getCurrentInstance();
  if (!vm) {
    throw new Error('useProp must be called within a setup function');
  }
  const proxy = vm.proxy as ComponentPublicInstance;
  const props = proxy?.$props as Record<string, T>;
  return computed(() => props?.[name]);
};
