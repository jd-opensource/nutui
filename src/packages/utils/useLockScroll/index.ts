import { body } from '@/packages/utils/util'
let count = 0

const CLSNAME = 'nut-overflow-hidden'

export const useLockScroll = (isLock: () => boolean) => {
  const lock = () => {
    if (isLock()) {
      try {
        !count && body.classList.add(CLSNAME)
        count++
      } catch (error) {
        console.warn('[NutUI] <useLockScroll>', error)
      }
    }
  }

  const unlock = () => {
    if (isLock() && count) {
      try {
        count--
        !count && body.classList.remove(CLSNAME)
      } catch (error) {
        console.warn('[NutUI] <unlock>', error)
      }
    }
  }

  return [lock, unlock]
}
