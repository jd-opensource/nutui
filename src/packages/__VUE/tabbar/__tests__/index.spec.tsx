import { mount } from '@vue/test-utils'
import { Tabbar, TabbarItem } from '@nutui/nutui'
import { nextTick, h, ref } from 'vue'
import { Home, Category, Find } from '@nutui/icons-vue'
import { mockGetBoundingClientRect, sleep } from '@/packages/utils/unit'
import { useRouter } from '@/packages/utils/useRoute'

vi.mock('@/packages/utils/useRoute')
;(useRouter as any).mockReturnValue({
  push: vi.fn()
})

test('Tabbar: should render tabbar when default', async () => {
  const wrapper = mount(() => {
    return (
      <Tabbar>
        <TabbarItem tab-title="首页" icon={h(Home)}></TabbarItem>
        <TabbarItem tab-title="分类" icon={h(Category)}></TabbarItem>
        <TabbarItem tab-title="发现" icon={h(Find)}></TabbarItem>
      </Tabbar>
    )
  })
  await nextTick()
  expect(wrapper.find('.nut-tabbar').exists()).toBe(true)
  expect(wrapper.find('.nut-tabbar-item').exists()).toBe(true)
  expect(wrapper.html()).toMatchSnapshot()
})

test('Tabbar: should render custom img when using img prop', async () => {
  const wrapper = mount(() => {
    return (
      <Tabbar>
        <TabbarItem tab-title="首页">
          {{
            icon: (props: any) =>
              props.active
                ? (
                    <img src="http://img13.360buyimg.com/uba/jfs/t1/29316/38/1115/3203/5c0f3d61E35d0c7da/9e557f2cb5c9dab6.jpg" />
                  )
                : (
                    <img src="http://img20.360buyimg.com/uba/jfs/t1/9996/36/8646/4833/5c0f3d61E7c1b7e0f/c98ad61124172e93.jpg" />
                  )
          }}
        </TabbarItem>
        <TabbarItem tab-title="分类">
          {{
            icon: (props: any) =>
              props.active
                ? (
                    <img src="http://img13.360buyimg.com/uba/jfs/t1/29316/38/1115/3203/5c0f3d61E35d0c7da/9e557f2cb5c9dab6.jpg" />
                  )
                : (
                    <img src="http://img20.360buyimg.com/uba/jfs/t1/9996/36/8646/4833/5c0f3d61E7c1b7e0f/c98ad61124172e93.jpg" />
                  )
          }}
        </TabbarItem>
      </Tabbar>
    )
  })
  await nextTick()
  const tabbarItemIcon: any = wrapper.findAll('.nut-tabbar-item_icon-box_icon img')
  expect(tabbarItemIcon[0].element.src).toContain('9e557f2cb5c9dab6')
  expect(tabbarItemIcon[1].element.src).toContain('c98ad61124172e93')
})

test('Tabbar: should render custom color and bage when using prop', async () => {
  const wrapper = mount(() => {
    return (
      <Tabbar unactive-color="grey" active-color="blue">
        <TabbarItem tab-title="首页" icon={h(Home)}></TabbarItem>
        <TabbarItem tab-title="分类" icon={h(Category)}></TabbarItem>
        <TabbarItem tab-title="发现" icon={h(Find)}></TabbarItem>
      </Tabbar>
    )
  })
  await nextTick()
  const tabbarItem: any = wrapper.findAll('.nut-tabbar-item')
  expect(tabbarItem.length).toBe(3)
  expect(tabbarItem[0].element.style.color).toEqual('blue')
  expect(tabbarItem[1].element.style.color).toEqual('grey')
  expect(wrapper.find<HTMLElement>('.nut-badge').exists()).toBe(true)
})

test('Tabbar: should render fixed element when using bottom prop', async () => {
  const wrapper = mount(Tabbar, {
    props: {
      bottom: true,
      safeAreaInsetBottom: true
    }
  })
  expect(wrapper.html()).toMatchSnapshot()
})
test('Tabbar: should match active tabbar by clcik', async () => {
  const wrapper = mount({
    setup() {
      const active = ref(0)
      return () => (
        <Tabbar unactive-color="grey" active-color="blue" v-model={active.value}>
          <TabbarItem tab-title="首页" icon={h(Home)}></TabbarItem>
          <TabbarItem tab-title="分类" icon={h(Category)}></TabbarItem>
          <TabbarItem tab-title="发现" icon={h(Find)}></TabbarItem>
        </Tabbar>
      )
    }
  })
  const tabbarItem: any = wrapper.findAll('.nut-tabbar-item')

  expect(tabbarItem[0].element.style.color).toEqual('blue')

  await tabbarItem[1].trigger('click')
  expect(tabbarItem[1].element.style.color).toEqual('blue')
  await tabbarItem[2].trigger('click')
  expect(tabbarItem[2].element.style.color).toEqual('blue')
})

test('Tabbar: should show sure emitted when click', async () => {
  const tabSwitch = vi.fn()
  const wrapper = mount(() => {
    return (
      <Tabbar onTabSwitch={tabSwitch}>
        <TabbarItem tab-title="首页" icon={h(Home)}></TabbarItem>
        <TabbarItem tab-title="分类" icon={h(Category)}></TabbarItem>
        <TabbarItem tab-title="发现" icon={h(Find)}></TabbarItem>
      </Tabbar>
    )
  })
  const tabbarItem: any = wrapper.findAll('.nut-tabbar-item')
  await tabbarItem[1].trigger('click')
  await nextTick()
  expect(tabSwitch).toBeCalled()
})

test('Tabbar: should render placeholder when using placeholder and bottom prop', async () => {
  const wrapper = mount(Tabbar, {
    props: {
      bottom: true,
      placeholder: true
    }
  })
  mockGetBoundingClientRect({ height: 40 })
  await sleep(500)
  expect(wrapper.html()).toMatchSnapshot()
})

test('Tabbar: should redirect when exist router and using to prop', async () => {
  const wrapper = mount(() => {
    return (
      <Tabbar>
        <TabbarItem tab-title="首页" icon={h(Home)} to="/home"></TabbarItem>
        <TabbarItem tab-title="分类" icon={h(Category)} to="/category"></TabbarItem>
        <TabbarItem tab-title="发现" icon={h(Find)} to="/category"></TabbarItem>
      </Tabbar>
    )
  })
  const tabbarItem: any = wrapper.findAll('.nut-tabbar-item')
  await tabbarItem[1].trigger('click')
  expect(useRouter().push).toHaveBeenCalledTimes(1)
  expect(useRouter().push).toHaveBeenCalledWith('/category')
})

test('Tabbar: should call replace when no router exist and using to prop', async () => {
  (useRouter as any).mockReturnValue(undefined)

  const wrapper = mount(() => {
    return (
      <Tabbar>
        <TabbarItem tab-title="首页" icon={h(Home)} to="/home"></TabbarItem>
        <TabbarItem tab-title="分类" icon={h(Category)} to="/category"></TabbarItem>
        <TabbarItem tab-title="发现" icon={h(Find)} to="/category"></TabbarItem>
      </Tabbar>
    )
  })
  location.replace = vi.fn()
  const tabbarItem: any = wrapper.findAll('.nut-tabbar-item')
  await tabbarItem[1].trigger('click')
  expect(location.replace).toHaveBeenCalledWith('/category')
})

test('Tabbar: should set window.location.href when using href prop', async () => {
  const wrapper = mount(() => {
    return (
      <Tabbar>
        <TabbarItem tab-title="首页" icon={h(Home)} href="/home"></TabbarItem>
        <TabbarItem tab-title="分类" icon={h(Category)} href="/category"></TabbarItem>
        <TabbarItem tab-title="发现" icon={h(Find)} href="/find"></TabbarItem>
      </Tabbar>
    )
  })
  const tabbarItem: any = wrapper.findAll('.nut-tabbar-item')
  await tabbarItem[1].trigger('click')
  expect(window.location.href).toMatch('/category')
})

test('Tabbar: props.beforeSwitch', async () => {
  const val = ref(0)
  const beforeSwitch = (_: any, index: string | number) => {
    return Number(index) % 2 === 0
  }
  const wrapper = mount(() => {
    return (
      <Tabbar v-model={val.value} beforeSwitch={beforeSwitch}>
        <TabbarItem tab-title="首页" icon={h(Home)}></TabbarItem>
        <TabbarItem tab-title="分类" icon={h(Category)}></TabbarItem>
        <TabbarItem tab-title="发现" icon={h(Find)}></TabbarItem>
      </Tabbar>
    )
  })
  const items = wrapper.findAll('.nut-tabbar-item')
  expect(items.length).toBe(3)
  items[2].trigger('click')
  await nextTick()
  expect(val.value).toBe(2)

  items[1].trigger('click')
  await nextTick()
  expect(val.value).toBe(2)
})
