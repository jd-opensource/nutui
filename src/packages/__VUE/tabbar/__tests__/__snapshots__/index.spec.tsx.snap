// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tabbar: should render fixed element when using bottom prop 1`] = `
"<div class="" style="">
  <view class="nut-tabbar nut-tabbar-bottom nut-tabbar-safebottom"></view>
</div>"
`;

exports[`Tabbar: should render placeholder when using placeholder and bottom prop 1`] = `
"<div class="nut-tabbar__placeholder" style="height: 40px;">
  <view class="nut-tabbar nut-tabbar-bottom"></view>
</div>"
`;

exports[`Tabbar: should render tabbar when default 1`] = `
"<div class="" style="">
  <view class="nut-tabbar">
    <div class="nut-tabbar-item">
      <view class="nut-badge">
        <view class="nut-badge__icon" style="top: 0px; right: 0px; z-index: 9; display: none;"></view>
        <view class="nut-tabbar-item_icon-box">
          <!--v-if-->
          <view><svg class="nut-popover-item-img nut-icon nut-icon-home" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1024 1024" role="presentation">
              <path d="M761.436 1024H262.564c-131.282 0-236.308-105.026-236.308-236.308V404.35c0-52.513 26.257-102.4 68.267-131.282L467.364 21.005c26.257-18.38 60.39-18.38 89.272 0l372.84 252.062c42.011 28.882 68.268 78.769 68.268 131.282v383.343c0 131.282-105.026 236.308-236.308 236.308zM139.159 338.708c-21.005 15.754-34.133 39.384-34.133 65.64v383.344c0 86.646 70.892 157.539 157.538 157.539h498.872c86.646 0 157.538-70.893 157.538-157.539V404.35c0-26.257-13.128-49.887-34.133-65.641L512 86.646 139.159 338.708zm451.61 396.471H433.231c-21.005 0-39.385-18.379-39.385-39.384s18.38-39.385 39.385-39.385h157.538c21.005 0 39.385 18.38 39.385 39.385s-18.38 39.384-39.385 39.384z" fill="currentColor" fill-opacity="0.9"></path>
            </svg></view>
          <view class="nut-tabbar-item_icon-box_nav-word">
            <view>首页</view>
          </view>
        </view>
        <view class="nut-badge__content nut-badge__content--sup" style="top: 0px; right: 0px; z-index: 9; display: none;"></view>
      </view>
    </div>
    <div class="nut-tabbar-item nut-tabbar-item__icon--unactive">
      <view class="nut-badge">
        <view class="nut-badge__icon" style="top: 0px; right: 0px; z-index: 9; display: none;"></view>
        <view class="nut-tabbar-item_icon-box">
          <!--v-if-->
          <view><svg class="nut-popover-item-img nut-icon nut-icon-category" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1024 1024" role="presentation">
              <path d="M307.2 468.114H160.914C73.143 468.114 0 394.971 0 307.2V160.914C0 73.143 73.143 0 160.914 0H307.2c87.771 0 160.914 73.143 160.914 160.914V307.2c0 87.771-73.143 160.914-160.914 160.914zM160.914 87.771c-40.96 0-73.143 32.183-73.143 73.143V307.2c0 40.96 32.183 73.143 73.143 73.143H307.2c40.96 0 73.143-32.183 73.143-73.143V160.914c0-40.96-32.183-73.143-73.143-73.143H160.914zM307.2 1024H160.914C73.143 1024 0 950.857 0 863.086V716.8c0-87.771 73.143-160.914 160.914-160.914H307.2c87.771 0 160.914 73.143 160.914 160.914v146.286c0 87.771-73.143 160.914-160.914 160.914zM160.914 643.657c-40.96 0-73.143 32.183-73.143 73.143v146.286c0 40.96 32.183 73.143 73.143 73.143H307.2c40.96 0 73.143-32.183 73.143-73.143V716.8c0-40.96-32.183-73.143-73.143-73.143H160.914zm629.029-175.543c-128.732 0-234.057-105.325-234.057-234.057S661.21 0 789.943 0 1024 105.326 1024 234.057 918.674 468.114 789.943 468.114zm0-380.343c-81.92 0-146.286 64.366-146.286 146.286s64.366 146.286 146.286 146.286 146.286-64.366 146.286-146.286S871.863 87.771 789.943 87.771zM863.086 1024H716.8c-87.771 0-160.914-73.143-160.914-160.914V716.8c0-87.771 73.143-160.914 160.914-160.914h146.286c87.771 0 160.914 73.143 160.914 160.914v146.286C1024 950.857 950.857 1024 863.086 1024zM716.8 643.657c-40.96 0-73.143 32.183-73.143 73.143v146.286c0 40.96 32.183 73.143 73.143 73.143h146.286c40.96 0 73.143-32.183 73.143-73.143V716.8c0-40.96-32.183-73.143-73.143-73.143H716.8z" fill="currentColor" fill-opacity="0.9"></path>
            </svg></view>
          <view class="nut-tabbar-item_icon-box_nav-word">
            <view>分类</view>
          </view>
        </view>
        <view class="nut-badge__content nut-badge__content--sup" style="top: 0px; right: 0px; z-index: 9; display: none;"></view>
      </view>
    </div>
    <div class="nut-tabbar-item nut-tabbar-item__icon--unactive">
      <view class="nut-badge">
        <view class="nut-badge__icon" style="top: 0px; right: 0px; z-index: 9; display: none;"></view>
        <view class="nut-tabbar-item_icon-box">
          <!--v-if-->
          <view><svg class="nut-popover-item-img nut-icon nut-icon-find" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1024 1024" role="presentation">
              <path d="M512 1024C230.4 1024 0 793.6 0 512S230.4 0 512 0s512 230.4 512 512-230.4 512-512 512zm0-930.91C281.6 93.09 93.09 281.6 93.09 512S281.6 930.91 512 930.91 930.91 742.4 930.91 512 742.4 93.09 512 93.09zM377.018 695.856c-20.945 0-39.563-11.637-46.545-32.582l-6.982-16.291 6.982-18.618 69.818-186.182c11.636-32.582 37.236-55.855 67.49-67.491l204.8-76.8L691.2 307.2c11.636 4.655 20.945 16.29 25.6 27.927l4.655 16.291-4.655 18.618-69.818 186.182c-11.637 32.582-37.237 55.855-67.491 67.491l-186.182 69.818c-4.654 2.328-11.636 2.328-16.29 2.328zm218.764-267.637-95.418 34.91c-6.982 2.327-11.637 6.981-13.964 13.963l-34.91 95.418 95.42-34.909c6.98-2.327 11.635-6.982 13.963-13.964l34.909-95.418z" fill="currentColor" fill-opacity="0.9"></path>
            </svg></view>
          <view class="nut-tabbar-item_icon-box_nav-word">
            <view>发现</view>
          </view>
        </view>
        <view class="nut-badge__content nut-badge__content--sup" style="top: 0px; right: 0px; z-index: 9; display: none;"></view>
      </view>
    </div>
  </view>
</div>"
`;
